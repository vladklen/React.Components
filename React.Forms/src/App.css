#root {
  max-width: 1280px;
  margin: 0 auto;
  padding: 2rem;
  text-align: center;
}

.logo {
  height: 6em;
  padding: 1.5em;
  will-change: filter;
  transition: filter 300ms;
}
.logo:hover {
  filter: drop-shadow(0 0 2em #646cffaa);
}
.logo.react:hover {
  filter: drop-shadow(0 0 2em #61dafbaa);
}

@keyframes logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

@media (prefers-reduced-motion: no-preference) {
  a:nth-of-type(2) .logo {
    animation: logo-spin infinite 20s linear;
  }
}

.card {
  padding: 2em;
}

.read-the-docs {
  color: #888;
}

form {
  display: flex;
  flex-direction: column;
}

form p {
  color: red;
}

main > div {
  border: 2px solid blue;
  border-radius: 10px;
}

.active {
  background-color: rgb(33, 56, 56);
}

.header-div {
  display: flex;
  flex-direction: row;
  gap: 20px;
  align-items: center;
}

.navbar ul {
  display: flex;
  flex-direction: row;
  gap: 20px;
  list-style: none;
}

a.active {
  color: aqua;
}

img {
  width: 100px;
  height: 100px;
}

footer {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  gap: 40px;
}
