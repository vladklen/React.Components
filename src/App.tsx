import { Component } from 'react';
import { ColorRing } from 'react-loader-spinner';
import MyInput from './components/UI/MyInput';
import MyButton from './components/UI/MyButton';
import { Card, CardProps } from './components/Ð¡ard';
import { ContentWrapper, SearchWrapper } from './components/UI/Styles';

interface IState {
  text: string;
  content: CardProps[];
  loading: boolean;
}

class App extends Component<unknown, IState> {
  constructor(props: unknown) {
    super(props);
    this.state = {
      text: localStorage.getItem('test') ?? '',
      content: [],
      loading: false,
    };
  }

  componentDidMount() {
    this.startSearch();
  }

  componentDidUpdate(): void {
    // eslint-disable-next-line react/destructuring-assignment
    console.log(this.state.content);
  }

  startSearch = async () => {
    const { text } = this.state;
    this.setState({ loading: true });
    const response = await fetch(
      `https://swapi.dev/api/people/?search=${text}`
    );
    const data = await response.json();
    this.setState({ content: data.results, loading: false });
  };

  handleInputChange = (event: { target: { value: string } }) => {
    const myValue = event.target.value;
    localStorage.setItem('test', myValue);
    this.setState({
      text: myValue,
    });
  };

  // eslint-disable-next-line class-methods-use-this
  throwError = async () => {
    try {
      await Promise.reject();
    } catch (error) {
      this.setState(() => {
        throw error;
      });
    }
  };

  render() {
    const { text, content, loading } = this.state;
    console.log(content);

    return (
      <div>
        <SearchWrapper>
          <MyInput message={text} change={this.handleInputChange} />
          <MyButton click={this.startSearch} color="blue" message="Search" />
          <MyButton click={this.throwError} color="red" message="Error Test" />
        </SearchWrapper>
        <h2>Results:</h2>
        <ContentWrapper>
          {loading && (
            <ColorRing
              visible
              height="80"
              width="80"
              ariaLabel="blocks-loading"
              wrapperStyle={{}}
              wrapperClass="blocks-wrapper"
              colors={['#e15b64', '#f47e60', '#f8b26a', '#abbd81', '#849b87']}
            />
          )}
          {content.length && !loading
            ? content.map((el) => (
                <Card
                  key={el.name}
                  name={el.name}
                  birth_year={el.birth_year}
                  height={el.height}
                  mass={el.mass}
                  gender={el.gender}
                />
              ))
            : !loading && <h3>Not found!</h3>}
        </ContentWrapper>
      </div>
    );
  }
}

export default App;
