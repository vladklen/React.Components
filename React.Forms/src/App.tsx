import './App.css';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import Layout from './Components/Layout';
import Home from './pages/Home';
import { PathConstants } from './routes/PathConstants';
import { FirstForm } from './pages/FirstForm';
import { SecondForm } from './pages/SecondForm';
import NoPage from './pages/NoPage';

export default function App() {
  return (
    <BrowserRouter>
      <Routes>
        <Route path="/" element={<Layout />}>
          <Route index element={<Home />} />
          <Route path={PathConstants.FIRST_FORM} element={<FirstForm />} />
          <Route path={PathConstants.SECOND_FORM} element={<SecondForm />} />
          <Route path="*" element={<NoPage />} />
        </Route>
      </Routes>
    </BrowserRouter>
  );
}
