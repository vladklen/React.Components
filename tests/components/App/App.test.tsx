import { render } from '@testing-library/react';
import { describe, test } from 'vitest';
import App from '../../../src/App';

describe('App tests', () => {
  test('renders app', () => {
    render(<App />);
  });
});
