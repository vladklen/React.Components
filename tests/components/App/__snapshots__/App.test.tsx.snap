// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App tests > renders app 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="sc-gtsqUy hcPYHT"
        >
          <input
            class="sc-iCoHzw jmVnpj"
            type="text"
            value=""
          />
          <button
            class="sc-gKAaef fTQyHl"
            data-testid="testButton"
            type="button"
          >
            Search
             
          </button>
        </div>
        <h2>
          Results:
        </h2>
        <div
          class="sc-bdnyFh fBMALl"
        >
          <div
            data-testid="card_1"
          >
            <div
              class="sc-eCAqax fHGSqI"
              data-testid="card"
            >
              <h3>
                Test title #1
              </h3>
              <img
                alt="Test title #1 preview"
                data-nimg="1"
                decoding="async"
                height="200"
                loading="lazy"
                src="/_next/image?url=url%3Apage1&w=384&q=75"
                srcset="/_next/image?url=url%3Apage1&w=256&q=75 1x, /_next/image?url=url%3Apage1&w=384&q=75 2x"
                style="color: transparent;"
                width="150"
              />
            </div>
          </div>
          <div
            data-testid="card_2"
          >
            <div
              class="sc-eCAqax fHGSqI"
              data-testid="card"
            >
              <h3>
                Test title #2
              </h3>
              <img
                alt="Test title #2 preview"
                data-nimg="1"
                decoding="async"
                height="200"
                loading="lazy"
                src="/_next/image?url=url%3Apage2&w=384&q=75"
                srcset="/_next/image?url=url%3Apage2&w=256&q=75 1x, /_next/image?url=url%3Apage2&w=384&q=75 2x"
                style="color: transparent;"
                width="150"
              />
            </div>
          </div>
          <div
            data-testid="card_3"
          >
            <div
              class="sc-eCAqax fHGSqI"
              data-testid="card"
            >
              <h3>
                Test title #3
              </h3>
              <img
                alt="Test title #3 preview"
                data-nimg="1"
                decoding="async"
                height="200"
                loading="lazy"
                src="/_next/image?url=url%3Apage3&w=384&q=75"
                srcset="/_next/image?url=url%3Apage3&w=256&q=75 1x, /_next/image?url=url%3Apage3&w=384&q=75 2x"
                style="color: transparent;"
                width="150"
              />
            </div>
          </div>
        </div>
        <div
          class="sc-hKFymg dNEsVI"
        >
          <select>
            <option
              value="1"
            >
              1
            </option>
            <option
              value="5"
            >
              5
            </option>
            <option
              value="10"
            >
              10
            </option>
          </select>
        </div>
        <ul
          aria-label="Pagination"
          class="sc-fujznN gOanmM"
          role="navigation"
        >
          <li
            class="previous disabled"
          >
            <a
              aria-disabled="true"
              aria-label="Previous page"
              class=" "
              rel="prev"
              role="button"
              tabindex="-1"
            >
              &lt;
            </a>
          </li>
          <li
            class="active"
          >
            <a
              aria-current="page"
              aria-label="Page 1 is your current page"
              rel="canonical"
              role="button"
              tabindex="-1"
            >
              1
            </a>
          </li>
          <li
            class="next disabled"
          >
            <a
              aria-disabled="true"
              aria-label="Next page"
              class=" "
              rel="next"
              role="button"
              tabindex="-1"
            >
              &gt;
            </a>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="sc-gtsqUy hcPYHT"
      >
        <input
          class="sc-iCoHzw jmVnpj"
          type="text"
          value=""
        />
        <button
          class="sc-gKAaef fTQyHl"
          data-testid="testButton"
          type="button"
        >
          Search
           
        </button>
      </div>
      <h2>
        Results:
      </h2>
      <div
        class="sc-bdnyFh fBMALl"
      >
        <div
          data-testid="card_1"
        >
          <div
            class="sc-eCAqax fHGSqI"
            data-testid="card"
          >
            <h3>
              Test title #1
            </h3>
            <img
              alt="Test title #1 preview"
              data-nimg="1"
              decoding="async"
              height="200"
              loading="lazy"
              src="/_next/image?url=url%3Apage1&w=384&q=75"
              srcset="/_next/image?url=url%3Apage1&w=256&q=75 1x, /_next/image?url=url%3Apage1&w=384&q=75 2x"
              style="color: transparent;"
              width="150"
            />
          </div>
        </div>
        <div
          data-testid="card_2"
        >
          <div
            class="sc-eCAqax fHGSqI"
            data-testid="card"
          >
            <h3>
              Test title #2
            </h3>
            <img
              alt="Test title #2 preview"
              data-nimg="1"
              decoding="async"
              height="200"
              loading="lazy"
              src="/_next/image?url=url%3Apage2&w=384&q=75"
              srcset="/_next/image?url=url%3Apage2&w=256&q=75 1x, /_next/image?url=url%3Apage2&w=384&q=75 2x"
              style="color: transparent;"
              width="150"
            />
          </div>
        </div>
        <div
          data-testid="card_3"
        >
          <div
            class="sc-eCAqax fHGSqI"
            data-testid="card"
          >
            <h3>
              Test title #3
            </h3>
            <img
              alt="Test title #3 preview"
              data-nimg="1"
              decoding="async"
              height="200"
              loading="lazy"
              src="/_next/image?url=url%3Apage3&w=384&q=75"
              srcset="/_next/image?url=url%3Apage3&w=256&q=75 1x, /_next/image?url=url%3Apage3&w=384&q=75 2x"
              style="color: transparent;"
              width="150"
            />
          </div>
        </div>
      </div>
      <div
        class="sc-hKFymg dNEsVI"
      >
        <select>
          <option
            value="1"
          >
            1
          </option>
          <option
            value="5"
          >
            5
          </option>
          <option
            value="10"
          >
            10
          </option>
        </select>
      </div>
      <ul
        aria-label="Pagination"
        class="sc-fujznN gOanmM"
        role="navigation"
      >
        <li
          class="previous disabled"
        >
          <a
            aria-disabled="true"
            aria-label="Previous page"
            class=" "
            rel="prev"
            role="button"
            tabindex="-1"
          >
            &lt;
          </a>
        </li>
        <li
          class="active"
        >
          <a
            aria-current="page"
            aria-label="Page 1 is your current page"
            rel="canonical"
            role="button"
            tabindex="-1"
          >
            1
          </a>
        </li>
        <li
          class="next disabled"
        >
          <a
            aria-disabled="true"
            aria-label="Next page"
            class=" "
            rel="next"
            role="button"
            tabindex="-1"
          >
            &gt;
          </a>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
